sdk/attestation/azure-security-attestation/src/private/attestation_client_private.hpp

```diff
@@ -204,12 +204,11 @@ namespace Azure { namespace Security { namespace Attestation { namespace _detail
       return Azure::Nullable<Models::AttestationSigner>();
     }
 
+    // Set the token body based on the bodyToSet parameter provided.
     template <typename Ty>
-    void SetTokenBody(
-        Azure::Core::Json::_internal::json const& jsonBody,
-        Azure::Nullable<Ty> bodyToSet)
+    void SetTokenBody(Azure::Core::Json::_internal::json const& jsonBody, Ty const* const bodyToSet)
     {
-      if (bodyToSet)
+      if (bodyToSet != nullptr)
```

Comment:
why do we need the `!= nullptr` ??   I feel to me like you can remove it

Reply:
`bodyToSet` is a raw pointer, not a smart pointer. IMHO when checking pointers, it's best not be explicit about null pointer checks and not rely on the implicit conversion from pointer to integer and then from integer to boolean.

`std::xxxx_ptr` provides an `operator bool` which allows customers to use the `if (bodyToSet)` construct safely. THe same is true for `Azure::Nullable`.

And you're right, it's not strictly necessary - the relevant pointer conversions are safe. But this is a coding style convention I have.

Also, to be clear: I *really* don't like the whole raw pointer thing. But Nullable<T> cannot be specialized on <void>.

Reply:
makes sense. Thank you